injected/src/features/ua-ch-brands.js

```diff
@@ -0,0 +1,130 @@
+import ContentFeature from '../content-feature';
+
+export default class UaChBrands extends ContentFeature {
+    constructor(featureName, importConfig, args) {
+        super(featureName, importConfig, args);
+
+        this.cachedBrands = null;
+        this.originalBrands = null;
+    }
+
+    init() {
+        const configuredBrands = this.getFeatureSetting('brands');
+        this.log.info(
+            'init() - configured brands from settings:',
+            configuredBrands ? configuredBrands.map((b) => `"${b.brand}" v${b.version}`).join(', ') : 'null',
+        );
+
+        if (!configuredBrands || configuredBrands.length === 0) {
+            this.log.info('No client hint brands correctly configured, feature disabled');
+            return;
+        }
+        this.shimUserAgentDataBrands();
+    }
+
+    /**
+     * Override navigator.userAgentData.brands to match the Sec-CH-UA header
+     */
+    shimUserAgentDataBrands() {
+        try {
+            // @ts-expect-error - userAgentData not yet standard
+            if (!navigator.userAgentData || !navigator.userAgentData.brands) {
+                this.log.info('shimUserAgentDataBrands - navigator.userAgentData not available');
+                return;
+            }
+
+            if (!this.originalBrands) {
+                // @ts-expect-error - userAgentData not yet standard
+                this.originalBrands = [...navigator.userAgentData.brands];
+                this.log.info('shimUserAgentDataBrands - captured original brands:', this.originalBrands.map(b => `"${b.brand}" v${b.version}`).join(', '));
+            }
+
+            if (this.cachedBrands) {
+                return;
+            }
+
+            const mutatedBrands = this.applyBrandMutations();
+
+            if (mutatedBrands) {
+                this.log.info('shimUserAgentDataBrands - about to apply override with:', mutatedBrands.map(b => `"${b.brand}" v${b.version}`).join(', '));
+                this.cachedBrands = mutatedBrands;
+                this.applyBrandsOverride(mutatedBrands);
+                this.log.info('shimUserAgentDataBrands - override applied, verify brands:', mutatedBrands.length, 'brands');
+            }
+        } catch (error) {
+            this.log.error('Error in shimUserAgentDataBrands:', error);
+        }
+    }
+
+    /**
+     * Find the GREASE brand value from original brands
+     * @returns {{brand: string, version: string}|null} - GREASE brand or null if not found
+     */
+    findGreaseBrand() {
+        if (!this.originalBrands || this.originalBrands.length === 0) {
+            return null;
+        }
+
+        return this.originalBrands.find((brand) => {
+            const name = brand.brand;
+            // Check if it starts with "Not" or " Not" or contains special chars
+            return name.trim().startsWith('Not') || /[^\w\s.]/.test(name);
```

Comment:
I think the special chars thing finds the GREASE brand as per: https://wicg.github.io/ua-client-hints/#create-arbitrary-brands-section but it might also pick up other values. Should we add an exclusion list from the config in case we match genuine values?

I think we can remove the "not" part, it's not part of the spec and could change.

What does Windows do here with the header?